src/test/gen-ts_test.ts

```diff
@@ -21,11 +22,21 @@ const goldens = path.join(__dirname, '..', '..', 'src', 'test', 'goldens');
 suite('generateDeclarations', () => {
   for (const fixture of fs.readdirSync(goldens)) {
     test(fixture, async () => {
-      const golden =
-          fs.readFileSync(path.join(goldens, fixture, 'expected.d.ts'));
-      const declarations =
-          await generateDeclarations(path.join(fixtures, fixture));
-      assert.equal(declarations, golden.toString());
+      // TODO Make these tests more async.
+      const actual = await generateDeclarations(path.join(fixtures, fixture));
+      const golden = readDirAsMap(path.join(goldens, fixture));
+      (assert as any).hasAllKeys(actual, [...golden.keys()]);
+      for (const filename of actual.keys()) {
+        assert.equal(actual.get(filename), golden.get(filename), filename);
+      }
     }).timeout(30000);  // These tests can take a long time.
```

Comment:
Wow, how long do these tests usually take?

Reply:
2-3 seconds, but a few times it was longer. Travis is sometimes resource constrained so I wanted to give a fairly big buffer.